core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatRelation.scala

```diff
@@ -113,7 +113,8 @@ class RowFormatRelation(
 
   override def buildUnsafeScan(requiredColumns: Array[String],
       filters: Array[Expression]): (RDD[Any], Seq[RDD[InternalRow]]) = {
-    val handledFilters = filters.flatMap(ExternalStoreUtils.handledFilter(_, indexedColumns))
+    val handledFilters = filters.flatMap(ExternalStoreUtils.handledFilter(_, indexedColumns
+      ++ pushdownPKColumns(filters)) )
```

Comment:
I am not sure that this is a right change. This is because for pk columns we have hash indexes i.e. the region itself. Other than equality how is it suppossed to work for greater than, less than etc. And equality is already handled.
Isn't it?

Reply:
Yes. and so if the filter condition has equality with correct columns ( so that they form primary key) then they should be part of handled filters. Pls take a look at bug , where I have described the exact issue. The check of handled/unhandled filters is done at two places.
1) To identify the unhandled filters
2) To identify the handled filters.
Both should work exactly same way. which was not the case. Unhandled filters wa staking into account primary key cols, while handled filter was not.